### ISSUE TITLE
Is it necessary to include climp in the linking options?

### ISSUE THREAD

**Issue by GitMensch** [LADSoft/OrangeC#340]
In this specific case (flex again) htonl and htons are referenced. Those are defined in winsock.h->win32\socket.h and seem to be part of climp.l

I'm just wondering if it is planned to specify climp.l on the linker options (I guess not).

**Comment by LADSoft** [comment]
right now if you are going to just call olink directly, you usually need to specify both clwin.l and climp.l on the linker command line.   climp.l is kind of an amalgamated version of the import libraries for windows system DLLs.   As such it is somewhat easier than the microsoft approach, which often has you including one import library for each DLL you want to reference.   For each identifier in climp.l that is used, olink/dlpe will create an entry in the executable's import table.

If you run occ and let it do the link it will automatically include climp.l when creating the linker command.

**Comment by GitMensch** [comment]
Hm, occ is run:

```
occ --nologo -o .libs/stage1flex.exe stage1flex-scan.o stage1flex-buf.o stage1flex-ccl.o stage1flex-dfa.o stage1flex-ecs.o stage1flex-filter.o stage1flex-gen.o stage1flex-main.o stage1flex-misc.o stage1flex-nfa.o stage1flex-options.o stage1flex-parse.o stage1flex-regex.o stage1flex-scanflags.o stage1flex-scanopt.o stage1flex-skel.o stage1flex-sym.o stage1flex-tables.o stage1flex-tables_shared.o stage1flex-tblcmp.o stage1flex-yylex.o ../lib/reallocarray.o
Error: Undefined External htonl in module ../../src/tables.c
Error: Undefined External htons in module ../../src/tables.c
```

**Comment by LADSoft** [comment]
hm I did find an issue where if you link against one of the DLL runtimes it wouldn't work but you don't seem to be doing that.    Will look into it...

**Comment by LADSoft** [comment]
ok there was nothing wrong, the deal is 'winsock.h' HAS to be included for berkely style sockets to work on windows.   Reason being that functions declared in winsock .h use the STDCALL calling convention, not the CDECL one.   Simply not including this header is enough to get it not to link.   Other than that it should work...    the 'no prototype diagnostic' might have been a hint that there was something going on here, but apparently when we did #216 we turned it off by default...
